### PR TITLE
[JENKINS-31531] Added Ownership plugin support

### PR DESCRIPTION
**Pull request by v1v**
https://issues.jenkins-ci.org/browse/JENKINS-31531

Added support for [Ownership plugin](https://wiki.jenkins-ci.org/display/JENKINS/Ownership+Plugin)

### CODE REVIEW

File: job-dsl-core/src/main/docs/examples/javaposse/jobdsl/dsl/helpers/properties/PropertiesContext/jobOwnership.groovy [hunk 1]
```diff
@@ -0,0 +1,10 @@
+job('example') {
+    properties {
+        jobOwnership {
+            primaryOwnerId('User_ID')
+            coownersIds('User1', 'User2')
```
Comment (daspilker) [inline job-dsl-core/src/main/docs/examples/javaposse/jobdsl/dsl/helpers/properties/PropertiesContext/jobOwnership.groovy]: use camel case, rename this to {{coOwnerIds}} everywhere

File: job-dsl-core/src/main/groovy/javaposse/jobdsl/dsl/helpers/properties/JobOwnershipContext.groovy [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import javaposse.jobdsl.dsl.Context
+
+class JobOwnershipContext implements Context {
+    String primaryOwnerId
+    List<String> coownersIds = []
+
+    /**
+     * Name of the primary owner of the job.
+     */
+    void primaryOwnerId(String primaryOwnerId = null) {
```
Comment (daspilker) [inline job-dsl-core/src/main/groovy/javaposse/jobdsl/dsl/helpers/properties/JobOwnershipContext.groovy]: remove the default value

**Comment by daspilker** [comment]
Please squash your commits when you are done.

**Comment by v1v** [comment]
Hi @daspilker ,

I've already squashed those commits

Cheers

**Comment by daspilker** [comment]
The co-owners are not accepted by Jenkins. Please fix that and test your code in Jenkins before submitting a PR.

**Comment by v1v** [comment]
Hi,

 I believe I did test it out before pushing those changes, I'll double check them now. Thanks for that

**Pull request merged**